src/Engine/GameAction.java

```diff
@@ -294,8 +296,45 @@ public GameEventQueue getEvents(MapMaster map)
 
           if( capture.willCapture() ) // If this will succeed, check if the CO will lose as a result.
           {
+            
+            boolean playerHasLost = false;
+            
             // Check if capturing this property will cause someone's defeat.
             if( (propertyType == TerrainType.HEADQUARTERS) && (null != captureLocation.getOwner()) )
+            {
+              playerHasLost = true;
+            }
+            
+            else if ((propertyType == TerrainType.LAB) && (null != captureLocation.getOwner())) 
+            {
+              int numLabs = 0;
+              int numHQs = 0;
+              Commander CO = captureLocation.getOwner();
+              for (XYCoord xy : CO.ownedProperties) 
+              {
+                Location loc = map.getLocation(xy);
+                if (loc.getEnvironment().terrainType == TerrainType.LAB ) 
+                {
+                  numLabs += 1;
+                  if (numLabs > 1) 
```

Comment:
This `if` is redundant. Since `numLabs` starts at zero, and you incremented it on the previous line, this check can never fail.
It doesn't hurt anything, but it doesn't help either.

Reply:
Nah, it's not redundant. He's checking that it's *greater* than 1, so he'll count up to 2 labs and then break.

Reply:
Oh. Right. Silly me.